openedx/openedx-platform
# feat: Add screen for partial feature based enrollment

## Description

Track selection, the screen which offers a verified certificate to a learner when they enroll, was implemented in #28418.

The text of that screen assumes the enrollment offers an audit track with full feature based enrollment (there is an access duration limit to the course, and there may be gated content not available to the learner).

Some enrollments may have partial feature based enrollment and may not restrict the audit track as stringently. This pull request adds a new screen that has more accurate text appropriate to partial feature based enrollment. 

<img width="1151" alt="TC02" src="https://user-images.githubusercontent.com/3790674/132081893-adf0ed1a-1306-44f8-98f0-e66c5e9c0c15.png">

## Supporting information

Closes: https://openedx.atlassian.net/browse/REV-2325

## Testing instructions

1.  Check course duration limits are enabled in the admin console prior to the creation and release date of the course being taken.
2.  Check gated content is enabled in the admin console prior to the creation and release date of the course being taken.
3.  Ensure the course has a verified and audit mode properly set up in the admin console.
4.  Using a test/non-staff user, enroll in the course.
5.  Verify user sees the full feature-based enrollment screen.
6.  Verify selecting the verified certificate track redirects to the payment page appropriately.
7.  Unenroll from the course, re-enroll in the course, and verify selecting the audit track redirects to the course page appropriately.
8.  Unenroll from the course.
9.  Disable gated content limits in the admin console.
10.  Repeat steps 4-8, verifying that this time the partial feature-based enrollment screen (screenshot above) shows.
11.  Repeat steps 9-10, this time disabling course duration limits and enabling gated content limits in the admin console. 

## Deadline

None.

## Other information

Out of scope to this pull request:
* Enterprise users
* Error handling for postbacks
